lib/barbeque/runner.rb

```diff
@@ -0,0 +1,55 @@
+require 'barbeque/config'
+require 'barbeque/exception_handler'
+require 'barbeque/execution_log'
+require 'barbeque/message_handler'
+require 'barbeque/message_queue'
+
+module Barbeque
+  # Part of barbeque-worker.
+  # Runner dequeues a message from {MessageQueue} (Amazon SQS) and dispatches
+  # it to message handler.
+
+  class Runner
+    DEFAULT_QUEUE = 'default'
+
+    def initialize(queue_name: ENV['BARBEQUE_QUEUE'] || DEFAULT_QUEUE)
+      @queue_name = queue_name
+    end
+
+    def run
+      keep_maximum_concurrent_executions
+
+      message = message_queue.dequeue
+      return unless message
```

Comment:
It's normal case.
`ReceiveMessage` returns empty response when there's no available message within `WaitTimeSeconds`.
http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-long-polling.html